src/main/java/net/sf/jabref/util/Util.java

```diff
@@ -498,12 +498,12 @@ public void run() {
                         }
                         if (!alreadyHas) {
                             foundAny = true;
-                            ExternalFileType type;
+                            Optional<ExternalFileType> type;
                             Optional<String> extension = FileUtil.getFileExtension(f);
                             if (extension.isPresent()) {
                                 type = ExternalFileTypes.getInstance().getExternalFileTypeByExt(extension.get());
```

Comment:
I just tried to copy the current behavior. One could also return
UnknownExternalFileType instead of empty() in the case where it is not
found. This is done in one (?) method, but I wasn't sure enough of the
exact mechanisms to do this right now. Drawback is maybe that one will need
to check instance type rather than is present, which somehow voids the use
of Optional...
Den 20 mar 2016 19:14 skrev "Tobias Diez" <notifications@github.com>:

> In src/main/java/net/sf/jabref/util/Util.java
> <https://github.com/JabRef/jabref/pull/1003#discussion_r56768868>:
>
> > @@ -498,12 +498,12 @@ public void run() {
> >                          }
> >                          if (!alreadyHas) {
> >                              foundAny = true;
> > -                            ExternalFileType type;
> > +                            Optional<ExternalFileType> type;
> >                              Optional<String> extension = FileUtil.getFileExtension(f);
> >                              if (extension.isPresent()) {
> >                                  type = ExternalFileTypes.getInstance().getExternalFileTypeByExt(extension.get());
>
> Would it make sense to use .orElse(UnknownExternalFileType) here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/JabRef/jabref/pull/1003/files/f8850929a0c8189645ace0583efb7e3dad0ef0e7#r56768868>
>